google/auth/headful.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Headful Credentials.
+Headful Credentials are …
+
+Example headful credential:
+{
+  "type": "external_account_authorized_user",
+  "audience": "//iam.googleapis.com/locations/global/workforcePools/$WORKFORCE_POOL_ID/providers/$PROVIDER_ID",
+  "refresh_token": "refreshToken",
+  "token_url": "https://sts.googleapis.com/v1/oauth/token",
+  "token_info_url": "https://sts.googleapis.com/v1/instrospect"
+  "client_id": "clientId",
+  "client_secret": "clientSecret"
+}
+"""
+
+_HEADFUL_JSON_TYPE = "external_account_authorized_user"
+
+class Credentials(external_account.Credentials):
+    """
+    """
+
+    def __init__(
+        self,
+        audience,
+        refresh_token,
+        token_url,
+        token_info_url,
+        client_id,
+        client_secret,
+        revoke_url="",
+        quota_project_id="",
+    ):
+        """Instantiates a headful credentials object."""
+        super(Credentials, self).__init__(
+            audience=audience,
+            subject_token_type=None,
+            token_url=token_url,
+            token_info_url=token_info_url,
+            revoke_url=revoke_url,
+            credential_source=None,
+            client_id=client_id,
+            client_secret=client_secret,
+            quota_project_id=quota_project_id,
+        )
+
+        self._refresh_token = refresh_token
+        self._token_info_url = token_info_url
+
+    @property
+    def info(self):
```

Comment:
Do we have a similar method in other creds?

maybe call this "to_info" if it is the exact opposite of "from_info" i.e. if we will be able to create this cred from the dictionary produced by this method.